detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerDetector.ts

```diff
@@ -51,21 +53,39 @@ export class DockerCGroupV1Detector implements Detector {
 
   private async _getContainerId(): Promise<string | undefined> {
     try {
-      const rawData = await DockerCGroupV1Detector.readFileAsync(
+      let rawData = await DockerDetector.readFileAsync(
         this.DEFAULT_CGROUP_PATH,
         this.UTF8_UNICODE
       );
-      const splitData = rawData.trim().split('\n');
+      let splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length >= this.CONTAINER_ID_LENGTH) {
           return str.substring(str.length - this.CONTAINER_ID_LENGTH);
         }
       }
+
+      // If this code is reached - we then check for the V2_PATH
+      rawData = await DockerDetector.readFileAsync(
+        this.DEFAULT_CGROUP_V2_PATH,
+        this.UTF8_UNICODE
+      );
+
+      splitData = rawData.trim().split('\n');
+      for (let str of splitData) {
+        if (str.includes(this.HOSTNAME)) {
+          str = str
```

Comment:
I think it will increase readability if instead of modifying `str` we introduce a new variable with a good name (is `containerId` what this string represents?): 
```js
const containerId = str.split('/').filter(...);
```

Reply:
sounds good.